apps/frontend/src/pages/root/root.tsx

```diff
@@ -1,6 +1,20 @@
+import { AppRoute } from "~/libs/enums/enums.js";
+import { useAppSelector, useEffect, useNavigate } from "~/libs/hooks/hooks.js";
+
 import { UserWheel } from "./components/components.js";
 
 const Root: React.FC = () => {
+	const navigate = useNavigate();
+	const { hasAnsweredQuizQuestions } = useAppSelector(({ auth }) => ({
+		hasAnsweredQuizQuestions: auth.user?.hasAnsweredQuizQuestions,
+	}));
+
+	useEffect(() => {
+		if (!hasAnsweredQuizQuestions) {
+			navigate(AppRoute.QUIZ);
+		}
+	}, [hasAnsweredQuizQuestions, navigate]);
```

Comment:
What if the user directly accesses another screen, like profile, or tasks? You should add this logic to ProtectedRoute component